### PR TITLE
[GraphQL/DB][EASY] (Re)move conversion to/from SuiAddress

### PR DESCRIPTION
**Pull request by amnn**
## Description

If we need conversion operators to/from `SuiAddress` and `Address`, move them into `sui_address.rs` or `address.rs` as appropriate, but in most cases, we don't need those operations.

## Test Plan

```
sui-graphql-rpc$ cargo nextest run
```

## Stack
- #15467 
- #15470 
- #15471 
- #15472 
- #15473
- #15474 
- #15475 
- #15484 
- #15485
- #15519

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]
neat

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs [hunk 1]
```diff
@@ -1185,8 +1185,7 @@ impl PgManager {
         name_service_config: &NameServiceConfig,
         address: SuiAddress,
     ) -> Result<Option<String>, Error> {
-        let reverse_record_id =
-            name_service_config.reverse_record_field_id(NativeSuiAddress::from(address));
+        let reverse_record_id = name_service_config.reverse_record_field_id(address.as_slice());
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:1188]: Why is this now a slice? The function takes a `SuiAddress`, and `address` is a `SuiAddress` so I am confused (without looking further into the actual types).

Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:1188]: this PR changes reverse_record_field_id to take &[u8]

File: crates/sui-json-rpc/src/name_service.rs [hunk 2]
```diff
@@ -96,11 +96,11 @@ impl NameServiceConfig {
         .unwrap()
     }
 
-    pub fn reverse_record_field_id(&self, address: SuiAddress) -> ObjectID {
+    pub fn reverse_record_field_id(&self, address: &[u8]) -> ObjectID {
```
Comment (stefan-mysten) [inline crates/sui-json-rpc/src/name_service.rs:99]: Oh, now I see, it's because here we pass a reference to a slice.

**Review by stefan-mysten — APPROVED** [review]

**Pull request merged**